### PR TITLE
UTOPIA-543: Added RichText module for better formatting

### PR DESCRIPTION
**Pull request by antsand**
This module addes rich text elements to a page using iframe The current functionality  uses execcommad which will soon be depriciated but is still widely used in most modern browsers.

Feature not included in this commit
TODO - Highlight box when rich text is active
     - Error and form validation highlighting
     - Activate richformat icon that is active

# Description

This PR includes the following proposed change(s):

- {List all the changes, if possible add the jira ticket #}

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Development Dependency Working Agreement
- [ ] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/frontend/src/components/common/RichText/index.tsx [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+import React from 'react';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { solid, regular, brands, icon } from '@fortawesome/fontawesome-svg-core/import.macro'
+import {
+    faBold, faItalic, faUnderline, faSubscript,
+    faSuperscript, faList, faListOl, faParagraph
+} from '@fortawesome/free-solid-svg-icons';
+
+class RichText extends React.Component {
```
Comment (Bottle7) [inline src/frontend/src/components/common/RichText/index.tsx:43]: Why are we using a class based component here? The rest of the frontend in function based.

File: src/frontend/src/components/common/RichText/index.tsx [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+        this.RichTextFrame.contentDocument.execCommand('superscript', false, null);
+    };
+    richTextOrderedList = function (this: RichText, event: object) {
+        this.RichTextFrame.contentDocument.execCommand('insertOrderedList', false, null);
+    };
+    richTextUnOrderedList = function (this: RichText, event: object) {
+        this.RichTextFrame.contentDocument.execCommand('insertUnorderedList', false, null);
+    };
+    richTextRemoveFormat = function (this: RichText, event: object) {
+        console.log('remove format');
```
Comment (kushal-arora-fw) [inline src/frontend/src/components/common/RichText/index.tsx:188]: `log` to be removed

File: src/frontend/src/components/common/RichText/index.tsx (hunk 1 above)
Comment (binggaofw) [reply src/frontend/src/components/common/RichText/index.tsx:43]: i strong suggest we use func based component

Comment (antsand) [reply src/frontend/src/components/common/RichText/index.tsx:43]: most of the documentation is class based. Just followed the document. In future interactions i can update it.

File: src/frontend/src/pages/PPQFormPage/PPQFormPage.tsx [hunk 3]
```diff
@@ -308,11 +309,12 @@ const PPQFormPage = () => {
...
                   : Messages.AdditionalInfoHeading.en}
               </h2>
-              <MDEditor
-                preview="edit"
-                value={description}
-                onChange={handleDescriptionChange}
-              />
+              <RichText
+                name="richText" /* this should be a unique name for each richtext instance */
+                key="Description"
```
[the remark this answers is not in the document]
Comment (antsand) [reply src/frontend/src/pages/PPQFormPage/PPQFormPage.tsx:314]: I would prefer to use name as it should be a string. Key sounds cryptic.
there is also a one to one mapping with html name and this name prop. Keeps things consistent.

**Pull request closed without merging**